File: day02/Passwords.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
)

func main() {
	dat, _ := ioutil.ReadFile(os.Args[1])
	lines := strings.Split(string(dat), "\n")
	valid := 0
	for i := 0; i < len(lines); i++ {
		line := strings.Split(lines[i], " ")
		policy := strings.Split(line[0], "-")
		policyMin, _ := strconv.Atoi(policy[0])
		policyMax, _ := strconv.Atoi(policy[1])
		character := line[1][0:1]
		password := line[2]
		occurrences := strings.Count(password, character)
		if occurrences >= policyMin && occurrences <= policyMax {
			valid++
		}
	}
	fmt.Printf("Valid password #: %v \n", valid)
}

File: day02/passwords.py
with open("input.txt") as f:
    lines = f.read().splitlines()

part1, part2 = 0, 0
for line in lines:
    parts = line.split(" ")
    policy = parts[0].split("-")
    character = parts[1][0]
    password = parts[-1]
    occurrences = password.count(character)
    if int(policy[0]) <= occurrences <= int(policy[-1]):
        part1 += 1
    if password[int(policy[0]) - 1] == character and not password[int(policy[-1]) - 1] == character:
        part2 += 1
    elif password[int(policy[-1]) - 1] == character and not password[int(policy[0]) - 1] == character:
        part2 += 1

print part1
print part2
